src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -572,6 +573,76 @@ private static bool HasApplicableConditionalMethod(OverloadResolutionResult<Meth
             return result;
         }
 
+        private BoundExpression BindDynamicArgToLocalFunction(
+            CSharpSyntaxNode syntax,
+            CSharpSyntaxNode expression,
+            string methodName,
+            BoundMethodGroup methodGroup,
+            DiagnosticBag diagnostics,
+            CSharpSyntaxNode queryClause,
```

Comment:
It turns out I couldn't find any situation in which query clause is non-null. When we bind query expressions we always create an outer lambda expression, which never seems to dig into the inner expression (which is what local function invocations always are).

I've added an `Assert` that this invariant is always true. Please let me know if you know of a case I missed. :)